### PR TITLE
QUICK-221 Make S3DatasetPackage public.

### PR DESCRIPTION
**Pull request by mminns**
Moved the class into the api package.
Tidied the API by removing data class definition.

### CODE REVIEW

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/dataset/S3DatasetPackage.kt [hunk 1]
```diff
@@ -1,16 +1,21 @@
-package com.atlassian.performance.tools.awsinfrastructure
+package com.atlassian.performance.tools.awsinfrastructure.api.dataset
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/dataset/S3DatasetPackage.kt]: We should mention new API in the changelog

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/dataset/S3DatasetPackage.kt [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
 import com.atlassian.performance.tools.infrastructure.api.dataset.DatasetPackage
 import com.atlassian.performance.tools.infrastructure.api.dataset.FileArchiver
 import com.atlassian.performance.tools.jvmtasks.api.TaskTimer.time
 import com.atlassian.performance.tools.ssh.api.SshConnection
 import java.time.Duration
 
-internal data class S3DatasetPackage(
+/**
+ * Downloads dataset from S3 and unzips on the remote machine.
+ * Requires valid AWS credentials.
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/dataset/S3DatasetPackage.kt]: Nice way to spread the tribal knowledge 👍

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/dataset/S3DatasetPackage.kt (hunk 1 above)
Comment (wyrzyk) [reply src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/dataset/S3DatasetPackage.kt]: We should add a bot to remind about that 😄 .

Comment (mminns) [reply src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/dataset/S3DatasetPackage.kt]: I've added an entry to the CHANGELOG.md.

Comment (mminns) [reply src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/dataset/S3DatasetPackage.kt]: I've added an entry to the CHANGELOG.md.

**Comment by mminns** [comment]
rebased on to current master.

**Pull request merged**